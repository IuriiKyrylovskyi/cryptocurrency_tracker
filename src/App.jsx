import React, { useState, useEffect } from 'react';
import { fetchCryptoList } from './gateway';
import Coins from './Coins';
import Search from './Search';

const App = () => {
  const [coins, setCoins] = useState([]);
  const [search, setSearch] = useState('');

  useEffect(() => {
    fetchCryptoList()
      .then(data => {
        setCoins(data);
      })
      .catch(error => new Error(error));
  }, []);

  console.log(coins);

  const handleChange = e => {
    setSearch(e.target.value);
  };

  const filteredCoins = coins.filter(coin =>
    coin.name.toLowerCase().includes(search.toLocaleLowerCase()),
  );

  return (
    <div className="main">
      <Search search={search} handleInput={handleChange} />
      <Coins coins={filteredCoins} />
    </div>
  );
};

export default App;
