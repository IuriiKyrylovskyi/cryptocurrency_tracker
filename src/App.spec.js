import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';

const setUp = () => shallow(<App />);

describe('should render App component', () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it('renders without crashing', () => {
    expect(component).toMatchSnapshot();
  });

  it('should contain .main wrapper', () => {
    const wrapper = component.find('.main');
    expect(wrapper.length).toBe(1);
  });
});
