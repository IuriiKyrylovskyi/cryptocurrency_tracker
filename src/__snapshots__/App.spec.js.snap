// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App component renders without crashing 1`] = `
<div
  className="main"
>
  <Search
    handleInput={[Function]}
  />
  <Coins
    coins={Array []}
  />
</div>
`;
